kie-server-parent/kie-server-controller/kie-server-controller-api/src/main/java/org/kie/server/controller/api/model/spec/ContainerConfig.java

```diff
@@ -19,8 +19,13 @@
 import javax.xml.bind.annotation.XmlAccessorType;
 import javax.xml.bind.annotation.XmlRootElement;
 
+import com.fasterxml.jackson.annotation.JsonSubTypes;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+
 @XmlAccessorType(XmlAccessType.FIELD)
 @XmlRootElement(name = "container-config")
+@JsonTypeInfo(use = JsonTypeInfo.Id.CLASS, include = JsonTypeInfo.As.WRAPPER_OBJECT)
+@JsonSubTypes({@JsonSubTypes.Type(value = ProcessConfig.class), @JsonSubTypes.Type(value = RuleConfig.class)})
```

Comment:
have you ran into any issues with default JSONMarshaller which pretty much does exactly the same thing - wrap the object with type information?

Reply:
@mswiderski exactly, so removing these classes from the minimal list for json and adding the type handling here, allows to keep the same content but avoid the type id handling exception in these different places that I tried to cover with unit tests.

Reply:
ok, I think it's fine ... was thinking if the extra jackson dependency will have any impact but for controller it should not be an issue...